FFmpegInterop/FFmpegInteropMSS.cpp

```diff
@@ -296,6 +297,31 @@ MediaPlaybackItem^ FFmpegInteropMSS::CreateMediaPlaybackItem(TimeSpan startTime,
 	}
 }
 
+IAsyncAction^ FFmpegInterop::FFmpegInteropMSS::ParseExternalSubtitleStream(IRandomAccessStream ^ stream)
+{
+	return create_async([this, stream]
+	{
+		auto config = ref new FFmpegInteropConfig();
+		config->IsExternalSubtitleParser = true;
+
+		auto externalSubsParser = FFmpegInteropMSS::createFromStreamInternal(stream, config, nullptr);
+		int readResult = 0;
+		while ((readResult = externalSubsParser->m_pReader->ReadPacket()) >= 0)
+		{
+			Concurrency::interruption_point();
+		}
```

Comment:
After the while loop we need to lock our mutex and unlock it again at the end of the method. Otherwise we could run into race conditions when modifying our internal lists.

Reply:
How would that work with the intreruption point though? I agree we need the lock.

Reply:
The first part (parsing) until end of the while loop can stay out of the lock. Only after that, we will modify our own status, so that part needs to be locked.